src/main/lrsql/system.clj

```diff
@@ -13,18 +14,20 @@
         (read-config profile)
         initial-sys ; init without configuration
         (component/system-map
+         ;; Logger is required by all other components so it initializes first
+         :logger     (logger/map->Logger {})
          :backend    (component/using
                       backend
-                      [])
+                      [:logger])
          :connection (component/using
                       (db/map->Connection {})
-                      [:backend])
+                      [:logger :backend])
          :lrs        (component/using
                       (lrs/map->LearningRecordStore {})
-                      [:connection :backend])
+                      [:logger :connection :backend])
          :webserver  (component/using
                       (webserver/map->Webserver {})
-                      [:lrs]))
+                      [:logger :lrs]))
```

Comment:
Didn't realize that you could have dependent components that _aren't_ record fields.

I'm not sure if you really need all these `using ... :logger` calls on the components above `:backend`, since they all build on top of each other.

Reply:
> Didn't realize that you could have dependent components that aren't record fields.

Yep, it's also an associative map. It's weird.

> I'm not sure if you really need all these using ... :logger calls on the components above :backend, since they all build on top of each other.

I don't love it, but since we want it to happen _before_ anything that could log at init in the other components, this seems to be the best way.

Reply:
Oh I see what you're saying, because everything else depends on backend already. Yeah, I can save some code that way I guess

Reply:
https://github.com/yetanalytics/lrsql/pull/238/commits/a5daeeaaf185be7b7169dd47f460e0cc302bfb8b